src/coreclr/jit/compiler.hpp

```diff
@@ -1324,9 +1324,10 @@ inline GenTreeIntCon* Compiler::gtNewIconHandleNode(size_t value, GenTreeFlags f
 
     GenTreeIntCon* node;
 #if defined(LATE_DISASM)
-    node = new (this, LargeOpOpcode()) GenTreeIntCon(TYP_I_IMPL, value, fields DEBUGARG(/*largeNode*/ true));
+    node = new (this, LargeOpOpcode())
+        GenTreeIntCon(gtGetTypeForIconFlags(flags), value, fields DEBUGARG(/*largeNode*/ true));
 #else
-    node             = new (this, GT_CNS_INT) GenTreeIntCon(TYP_I_IMPL, value, fields);
+    node             = new (this, GT_CNS_INT) GenTreeIntCon(gtGetTypeForIconFlags(flags), value, fields);
```

Comment:
Unrelated, but any idea why this needs a large node for `LATE_DISASM`?

Reply:
Pretty sure it'd dead code.

Reply:
We revived `LATE_DISASM` for use in SVE emitter unit tests in #96286, so it's not dead anymore.

Reply:
Yes, but I don't think this large opcode `ifdef`ing is still needed. It seems in the past `gtCompileTimeHandle` was part of the requirement for a "large" node:

https://github.com/dotnet/runtime/blob/24d149c7e761293f666ed8e1bb06675c6f65b4ca/src/coreclr/jit/morph.cpp#L12663-L12669